### PR TITLE
Fix large memory use caused by the use of multiprocessing.imap_unordered.

### PR DESCRIPTION
**Pull request by Bodong-Yang**
# Introduction
Previously, I introduce `multiprocessing.imap_unordered` to accelerate the `RegularInf` parsing.
But after small scale tests with `memory profiling` and `timeit`, I found that comparing to the time it saves(around 10seconds), it consume 200MB~ extra memory, which is not so efficient.
This PR dips the use of `multiprocessing.imap_unordered`, and also reduce some memory footprint for `RegularInf` class.

# Test and result
For one mainecu and one subecu environment, the memory usage on the mainecu drops from `867M~` to `664M~`.

### CODE REVIEW

**Comment by Bodong-Yang** [comment]
( I use the previous PR as base to commit this PR, so the commits for this PR are large in number, but it is not a problem as the previous PR has been merged.

File: app/ota_client.py [hunk 1]
```diff
@@ -297,6 +306,8 @@ class PersistentInf(_BaseInf):
     Persistent file information class
     """
 
+    __slots__ = ["path"]
+
```
Comment (obi-t4) [inline app/ota_client.py:310]: adding `del self._left` in the `__init__` is nice to have.

File: app/ota_client.py [hunk 2]
```diff
@@ -246,44 +248,51 @@ def __init__(self, info: str):
...
 
     _pattern = re.compile(r"'(?P<link>.+)((?<!\')',')(?P<target>.+)'")
 
+    __slots__ = ["slink", "srcpath"]
+
     def __init__(self, info):
         super().__init__(info)
         res = self._pattern.match(self._left)
         assert res is not None
+        del self._left
```
Comment (obi-t4) [inline app/ota_client.py:273]: add blank line before `del self._left` same as DirectoryInf and RegularInf. Or remove all blank lines before it for each class.

**Review by obi-t4 — COMMENTED** [review]
I left some comments, but basically, it is good.

**Comment by obi-t4** [comment]
@Bodong-Yang Commit history in this PR seems to be strange. Merge from main or some operations seem to be required.

**Comment by Bodong-Yang** [comment]
Superseded by #118.

File: app/ota_client.py [hunk 3]
```diff
@@ -231,12 +231,14 @@ class _BaseInf:
...
 
+    __slots__ = ["mode", "uid", "gid", "_left"]
+
     @staticmethod
     def de_escape(s: str) -> str:
         return s.replace(r"'\''", r"'")
 
     def __init__(self, info: str):
-        match_res: re.Match = self._base_pattern.match(info.strip("\n"))
+        match_res: re.Match = self._base_pattern.match(info.strip())
```
Comment (obi-t4) [inline app/ota_client.py:241]: @Bodong-Yang This strip() strips not only `\n` but also ` `?
This should be reverted?

**Pull request closed without merging**